prometheus.lua

```diff
@@ -829,25 +820,47 @@ function Prometheus:metric_data()
   local seen_metrics = {}
   local output = {}
   for _, key in ipairs(keys) do
+    utils.yield(true)
+
     local value, err = self.dict:get(key)
     if value then
-      local short_name = short_metric_name(key)
-      if not seen_metrics[short_name] then
+      local match, err = ngx_re_match(key, METRICS_KEY_REGEX, "jo")
+
+      if err then
+        ngx.log(ngx.ERR, "failed to match regex: ", err)
+        return
+      end
+
+      local has_seen = false
+      for seen_metric, _ in pairs(seen_metrics) do
+        if utils.has_prefix(key ,seen_metric) then
+          has_seen = true
+          break
+        end
+      end
```

Comment:
It seems that this might result in false positives if one metric name is a prefix of another: for example, if someone defines two metrics like `http_requests` and `http_requests_latencies`. Even though their short metric names are different, if `http_requests` gets added to the `seen_metrics` first, `HELP` and `TYPE` for `http_requests_latencies` will not get printed.

Reply:
yes, you are right, so I restored `fix_histogram_bucket_labels`